indexing-service/src/test/java/org/apache/druid/indexing/common/task/CompactionTaskTest.java

```diff
@@ -1512,7 +1512,7 @@ public void testQueryGranularityAndNullSegmentGranularity() throws IOException
         AGGREGATORS.stream().map(AggregatorFactory::getCombiningFactory).collect(Collectors.toList()),
         SEGMENT_INTERVALS,
         Granularities.MONTH,
-        new PeriodGranularity(Period.months(3), null, null),
```

Comment:
hmm, I wonder what it meant previously when query granularity was 3 months, while the segment granularity was 1 month? Was this an oversight not handled in this test and code : https://github.com/apache/druid/blob/master/indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java#L657-L660

Reply:
Yes, it could break existing compaction configurations.
My observation is that for replacing jobs such as compaction, if we specify a coarser query granularity like ALL and segment granularity as DAY,
1. We process the data as if the segment granularity and query granularity were both DAY.
2. The segment metadata retains query granularity as ALL and segment granularity as DAY.